tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -327,32 +435,38 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
     return OkStatus();
   }
 
-  StatusOr<bool> FuseVectorBiasAdd(HloInstruction *instr,
+  StatusOr<bool> FuseVectorBiasAdd(HloInstruction *add,
                                    HloInstruction *broadcast_bias,
-                                   HloInstruction *matmul) {
-    TF_RET_CHECK(broadcast_bias->shape() == matmul->shape());
+                                   HloInstruction *gemm,
+                                   HloInstruction *slice = nullptr) {
+    TF_RET_CHECK(broadcast_bias->shape() ==
+                 (slice ? slice->shape() : gemm->shape()));
```

Comment:
I think this needs to be `ShapeUtil::Compatible` rather than `==`.  In passes that run before layout assignment, one needs to ignore shape layouts, which may or may not be present, but are never meaningful.